Assets/GothicVR/Scripts/Manager/AudioManager.cs

```diff
@@ -0,0 +1,127 @@
+using UnityEngine;
+using System.Collections.Generic;
+using GVR.Util;
+
+namespace GVR.Manager
+{
+    public class AudioSourceManager : SingletonBehaviour<AudioSourceManager>
+    {
+        private Dictionary<GameObject, AudioSource> audioSources = new Dictionary<GameObject, AudioSource>();
+        private AudioListener audioListener = null;
+
+        private Vector3Int playerCell;
+
+        private bool IsAudioListenerObtained
+        {
+            get
+            {
+                return audioListener != null;
+            }
+        }
+
+        // Spatial partitioning variables
+        private Dictionary<Vector3Int, List<GameObject>> gridCells = new Dictionary<Vector3Int, List<GameObject>>();
+        private float cellSize = 50f; // Cell size in meters
+
+        private void Create()
+        {
+            // Initialize the grid cells
+            gridCells.Clear();
+        }
+
+        private void Update()
```

Comment:
performance seems to be the same for both approaches, but opted to have the function called once every half a second, in case it would have a performance hit on quest/pico for multiple sounds sources if it were executed every frame